src/Compilers/CSharp/Test/Symbol/Symbols/ModuleInitializersTests.cs

```diff
@@ -139,6 +141,131 @@ namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : S
 Program.Main");
         }
 
+        [Fact]
+        public void SingleCallIsGeneratedWhenMethodIsMarkedTwice()
+        {
+            string source = @"
+using System;
+using System.Runtime.CompilerServices;
+
+class C
+{
+    [ModuleInitializer, ModuleInitializer]
+    internal static void M() => Console.WriteLine(""C.M"");
+}
+
+class Program 
+{
+    static void Main() { }
+}
+
+namespace System.Runtime.CompilerServices
+{
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
+    class ModuleInitializerAttribute : System.Attribute { } 
+}
+";
+            CompileAndVerify(source, parseOptions: s_parseOptions, expectedOutput: "C.M");
+        }
+
+        [Fact]
+        public void AttributeCanBeAppliedWithinItsOwnDefinition()
+        {
+            string source = @"
+using System;
+
+class Program 
+{
+    static void Main() => Console.WriteLine(""Program.Main"");
+}
+
+namespace System.Runtime.CompilerServices
+{
+    class ModuleInitializerAttribute : System.Attribute 
+    { 
+        [ModuleInitializer]
+        internal static void M() => Console.WriteLine(""ModuleInitializerAttribute.M"");
+    } 
+}
+";
+            CompileAndVerify(source, parseOptions: s_parseOptions, expectedOutput: @"
+ModuleInitializerAttribute.M
+Program.Main");
+        }
```

Comment:
Consider testing what happens when the `class ModuleInitializerAttribute` has a `[Conditional]` attribute on it. Testing when the condition is met and when the condition is not met.

Consider testing what happens when the static method targeted by `[ModuleInitializer]` also has a `[Conditional]` attribute on it. Similarly test when the condition is met and when it is not met.

Reply:
Ooh, great thought.

Reply:
@RikkiGibson This caught a bug. Updated.